docs/content/developer/workshops/core/I-introduction-to-smart-contracts.mdx

```diff
@@ -0,0 +1,363 @@
+│  │   ┌──────────┬────────────────────────────────────────────────────────────────────┐               │
+│  │   │ amount   │ <amount>                                                           │               │
+│  │   ├──────────┼────────────────────────────────────────────────────────────────────┤               │
+│  │   │ grant_id │ 0x<grant_object_id>                                                │               │
+│  │   ├──────────┼────────────────────────────────────────────────────────────────────┤               │
+│  │   │ student  │ 0x<student_address>                                                │               │
+│  │   └──────────┴────────────────────────────────────────────────────────────────────┘               │
+│  └──                                                                                                 │
+```
+
+## Verify what happened
+
+```bash
+iota client objects --owner "$STUDENT_ADDR"
+iota client object --id <GRANT_OBJECT_ID>
+```
+
+This confirms that a `Grant` object now exists and is owned by the student. For events, use your indexer of choice and the [Using Events](../../iota-101/using-events.mdx) patterns. Ownership concepts recap: [Object Ownership](../../iota-101/objects/object-ownership/object-ownership.mdx).
+
+## Handy script (optional)
+
+```bash
+#!/usr/bin/env bash
+set -euo pipefail
+
+: "${PKG_ID:?set PKG_ID}"
+: "${ADMIN_CAP_ID:?set ADMIN_CAP_ID}"
+: "${STUDENT_ADDR:?set STUDENT_ADDR}"
+AMOUNT="${AMOUNT:-1000}"
+
+iota client call \
+  --package "$PKG_ID" \
+  --module grant \
+  --function mint \
+  --args "$STUDENT_ADDR" "$AMOUNT" "$ADMIN_CAP_ID" \
+  --gas-budget 100000000
+```
+
+## What we learned
+- Summary of what we built.
+- To further practise, we suggest trying X,Y (Extension tasks)
+- Add hints or extension tasks with the objective of deepening understanding.
+
+## Common pitfalls
+
+:::caution
+- Network: ensure your CLI is connected to Testnet (`iota client envs`; switch with `iota client switch --env testnet`).
+- Funding: request test tokens via faucet before publishing/calling (`iota client faucet --address $PUBLISHER_ADDR`).
+- First publish may rewrite addresses — re-check `Move.toml` and rebuild if you changed named addresses.
+- Make sure you pass the correct `ADMIN_CAP_ID` to `mint`.
+- Increase `--gas-budget` if you see out-of-gas failures.
+- Double-check signer and argument order for `iota client call`.
+:::
+
+## Bring this to Workshop II
+
+Keep these handy:
+
+- `PKG_ID`
+- `ADMIN_CAP_ID`
+- At least one `GRANT_OBJECT_ID`
+- Publisher and student addresses
+
+## Further reading
+
+- [Get Address](../../getting-started/get-address.mdx)
+- [Programmable Transaction Basics](../../ts-sdk/typescript/transaction-building/basics.mdx)
+- [Move Overview: Abilities](../../references/move/abilities.mdx)
+- [CLI Cheatsheet](../../references/cli/cheatsheet.mdx)
+
+
+
+## Further notes
```

Comment:
We may want to delete further notes, as it seems like for internal usage

Reply:
Good catch